sklearn/utils/estimator_checks.py

```diff
@@ -277,6 +277,7 @@ def _yield_outliers_checks(estimator):
         # test if NotFittedError is raised
         if _safe_tags(estimator, key="requires_fit"):
             yield check_estimators_unfitted
+    yield check_non_transformer_estimators_n_iter
```

Comment:
interesting that this wasn't here, where was the check before then?

Reply:
I think the number of iterations from the outlier detectors were not checked before. Otherwise `OneClassSVM` would have failed the test since it has an attribute `max_iter` but it does not have `n_iter`